contrib/learn_resource_weights.py

```diff
@@ -0,0 +1,167 @@
+                        help='ignore benchmarks with a runtime below this threshold')
+    parser.add_argument('--mult', type=int, default=1000,
+                        help='multiply running times with this factor for regression')
+
+    return parser.parse_args()
+
+
+args = parse_commandline()
+
+if args.verbose:
+    logging.getLogger().setLevel(level=logging.DEBUG)
+else:
+    logging.getLogger().setLevel(level=logging.INFO)
+
+
+def load_zipped_json(filename):
+    """Load data from a gziped json file"""
+    import gzip
+    with gzip.GzipFile(args.json, 'r') as fin:
+        return json.loads(fin.read().decode('utf-8'))
+
+
+def save_zipped_json(filename, data):
+    """Store data to a gziped json file"""
+    import gzip
+    with gzip.GzipFile(args.json, 'w') as fout:
+        fout.write(json.dumps(data).encode('utf-8'))
+
+
+def get_sorted_values(data):
+    """Transform [['name', value], ...] to [value, ...] sorted by names"""
+    return [d[1] for d in sorted(data)]
+
+
+if args.command == 'parse':
+    if args.basedir == None:
+        raise Exception('Specify basedir for parsing!')
+    import glob
+    import re
+    filename_re = re.compile(
+        '.*_incremental_([A-Z_]+/[0-9a-zA-Z_=\\./-]+\\.smt2)/output\\.log')
+    resource_re = re.compile('resource::([^,]+), ([0-9]+)')
+    result_re = re.compile('driver::sat/unsat, ([a-z]+)')
+    totaltime_re = re.compile('driver::totalTime, ([0-9\\.]+)')
+
+    logging.info('Parsing files from {}'.format(args.basedir))
+    data = {}
+    failed = 0
+    for file in glob.iglob('{}/**/output.log'.format(args.basedir), recursive=True):
+        content = open(file).read()
+        try:
+            filename = filename_re.match(file).group(1)
+            r = resource_re.findall(content)
+            r = list(map(lambda x: (x[0], int(x[1])), r))
+            data[filename] = {
+                'resources': r,
+                'result': result_re.search(content).group(1),
+                'time': float(totaltime_re.search(content).group(1)),
+            }
+        except Exception as e:
+            logging.debug('Failed to parse {}: {}'.format(file, e))
+            failed += 1
+
+    if failed > 0:
+        logging.info('Failed to parse {} out of {} files'.format(
+            failed, failed + len(data)))
+    logging.info('Dumping data to {}'.format(args.json))
+    save_zipped_json(args.json, data)
+
+elif args.command == 'analyze':
+    logging.info('Loading data from {}'.format(args.json))
+    data = load_zipped_json(args.json)
+
+    logging.info('Extracting resources')
+    resources = set()
+    for f in data:
+        spend_something = False
+        for r in data[f]['resources']:
+            if r[0].startswith('Bv') and r[1] > 0:
+                spend_something = True
+            resources.add(r[0])
+        if spend_something and f.startswith('QF_BV/') and data[f]['time'] < 0.004:
+            print(f)
```

Comment:
Am I missing something or does this look like a leftover of a debugging session? :)